scheduler/core/programprovider/ocs/ocsprogramprovider.py

```diff
@@ -508,19 +508,26 @@ def parse_nonsidereal_target(self, data: dict) -> NonsiderealTarget:
     def _parse_instrument(data: dict) -> str:
         """Get the instrument name"""
         fpu = None
+        instrument = None
 
-        inst_key = OcsProgramProvider._AtomKeys.INSTRUMENT
-        # Visitor instrument names are in OcsProgramProvider._AtomKeys.INST_NAME
-        if OcsProgramProvider._AtomKeys.INST_NAME in data:
-            inst_key = OcsProgramProvider._AtomKeys.INST_NAME
-        instrument = data[inst_key].split(' ')[0]
+        for step in data:
+            # Ignore acquisitions
+            if step[OcsProgramProvider._AtomKeys.OBS_CLASS].upper() not in [ObservationClass.ACQ.name,
+                                                                            ObservationClass.ACQCAL.name]:
+                inst_key = OcsProgramProvider._AtomKeys.INSTRUMENT
+                # Visitor instrument names are in OcsProgramProvider._AtomKeys.INST_NAME
+                if OcsProgramProvider._AtomKeys.INST_NAME in step:
+                    inst_key = OcsProgramProvider._AtomKeys.INST_NAME
+                instrument = step[inst_key].split(' ')[0]
 
-        if instrument in OcsProgramProvider.FPU_FOR_INSTRUMENT:
-            if OcsProgramProvider.FPU_FOR_INSTRUMENT[instrument] in data:
-                fpu = data[OcsProgramProvider.FPU_FOR_INSTRUMENT[instrument]]
+                if instrument in OcsProgramProvider.FPU_FOR_INSTRUMENT:
+                    if OcsProgramProvider.FPU_FOR_INSTRUMENT[instrument] in step:
```

Comment:
It seems like these if statements should be combined?
```python
if instrument in OcsProgramProvider.FPU_FOR_INSTRUMENT and OcsProgramProvider.FPU_FOR_INSTRUMENT[instrument] in step:
    fpu = step[OcsProgramProvider.FPU_FOR_INSTRUMENT[instrument]]
```
Not really a big deal, though.

Reply:
We could also probably simplify this down for `instrument` logic  and `fpu` logic by using the `dict` method `get(key)`, which returns `None` if `key` is not defined and the value of `key` otherwise. Just an FYI for future times this comes up.